sdcm/utils/validators/iotune.py

```diff
@@ -0,0 +1,102 @@
+
+    def _format_results_to_console(self):
+        preset_properties = self._read_io_properties()
+        preset_disk = next(iter(preset_properties.get("disks")))
+
+        for disk in self.node_io_properties.get("disks"):
+            disk_copy = {**disk}
+            mountpoint = disk_copy.pop("mountpoint")
+            LOGGER.info("Disk performance values validation - testing %s", mountpoint)
+            for key, val in disk_copy.items():
+                preset_val = preset_disk.get(key)
+                diff = (val / preset_val - 1) * 100
+                LOGGER.info("[%s] %s: %s (%.0f)", mountpoint, key, val, diff)
+
+    def _submit_results_to_argus(self):
+        def bottom_limit(metric_val, threshold=0.15):
+            metric_val - (metric_val * threshold)
+
+        if not self.argus_client:
+            LOGGER.warning("Will not submit to argus - no client initialized")
+            return
+        preset_io_props = self._read_io_properties()
+        preset_disk: IOTuneDiskParams = next(iter(preset_io_props.get("disks", [])), None)
+        if not preset_disk:
+            LOGGER.warning("Unable to continue - node should have io_properties.yaml, but it doesn't.")
+            return
+
+        class IOPropertiesResultsTable(GenericResultTable):
+            class Meta:
+                name = f"{self.params.get('cluster_backend')} - {self.node.db_node_instance_type} Disk Performance"
+                description = "io_properties.yaml comparison with live data"
+                Columns = [
+                    ColumnMetadata(name="read_iops", unit="iops", type=ResultType.INTEGER, higher_is_better=True),
+                    ColumnMetadata(name="read_bandwidth", unit="bps", type=ResultType.INTEGER, higher_is_better=True),
+                    ColumnMetadata(name="write_iops", unit="iops", type=ResultType.INTEGER, higher_is_better=True),
+                    ColumnMetadata(name="write_bandwidth", unit="bps", type=ResultType.INTEGER, higher_is_better=True),
+                ]
+
+                ValidationRules = {
+                    "read_iops": ValidationRule(best_pct=10, fixed_limit=bottom_limit(preset_disk.get("read_iops"))),
+                    "read_bandwidth": ValidationRule(best_pct=10, fixed_limit=bottom_limit(preset_disk.get("read_iops"))),
+                    "write_iops": ValidationRule(best_pct=10, fixed_limit=bottom_limit(preset_disk.get("read_iops"))),
+                    "write_bandwidth": ValidationRule(best_pct=10, fixed_limit=bottom_limit(preset_disk.get("read_iops"))),
+                }
+
+        tested_disk: IOTuneDiskParams = next(iter(self.node_io_properties.get("disks", [])))
+        tested_mountpoint = tested_disk.pop("mountpoint")
+        if tested_mountpoint != preset_disk["mountpoint"]:
+            LOGGER.warning("Disks differ - probably a mistake: %s vs %s, will not submit iotune results",
+                           tested_mountpoint, preset_disk["mountpoint"])
+            return
```

Comment:
I think we shouldn't fail silently here. Better raise error event

Reply:
Will do

Reply:
Done